### PR TITLE
Backport rust 1.26 from master.

### PR DESCRIPTION
**Pull request by shlevy**
As of 47529594e4dbc02ab31d30df608bff0cc991b869.

### CODE REVIEW

**Comment by shlevy** [comment]
cc @Havvy

**Comment by shlevy** [comment]
@GrahamcOfBorg build rust_1_26.rustc

**Comment by shlevy** [comment]
(build still running locally)

**Comment by shlevy** [comment]
@GrahamcOfBorg build rust_1_26.cargo

**Comment by shlevy** [comment]
:sob: Guess I'll have to wait for my puny laptop to finish

File: pkgs/development/compilers/rust/1.26/binaryBuild.nix [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        install_name_tool -change /usr/lib/libcurl.4.dylib '${stdenv.lib.getLib curl}/lib/libcurl.4.dylib' "$out/bin/cargo"
+        for f in $out/lib/lib*.dylib; do
+          install_name_tool -change /usr/lib/libresolv.9.dylib '${darwin.libresolv}/lib/libresolv.9.dylib' "$f"
+        done
+      ''}
+
+      # Do NOT, I repeat, DO NOT use `wrapProgram` on $out/bin/rustc
+      # (or similar) here. It causes strange effects where rustc loads
+      # the wrong libraries in a bootstrap-build causing failures that
+      # are very hard to track dow. For details, see
```
Comment (Havvy) [inline pkgs/development/compilers/rust/1.26/binaryBuild.nix]: "dow" -> "down"

Comment (shlevy) [reply pkgs/development/compilers/rust/1.26/binaryBuild.nix]: Thanks, fixed in https://github.com/NixOS/nixpkgs/commit/afad9983e319be2a9c732f53a41cd092b548a0fd

**Review by LnL7 — APPROVED** [review]
Fine by me, assuming the build doesn't depend on something else from master.

**Review by wizeman — APPROVED** [review]
I haven't been keeping up with Rust packaging, but assuming it builds and works, it LGTM :thumbsup:

**Pull request merged**